yktrace/src/swt.rs

```diff
@@ -2,32 +2,42 @@
 
 use super::{SirTrace, ThreadTracer, ThreadTracerImpl};
 use crate::errors::InvalidTraceError;
-use core::yk::{swt, SirLoc};
+use std::convert::TryFrom;
+use core::yk::{swt, SirLoc as CoreSirLoc};
+use crate::SirLoc;
 use libc;
-use std::ops::Drop;
 
 /// A trace collected via software tracing.
 /// Since the trace is a heap-allocated C buffer, we represent it as a pointer and a length.
 #[derive(Debug)]
 struct SWTSirTrace {
-    buf: *mut SirLoc,
-    len: usize
+    locs: Vec<SirLoc>,
+}
+
+impl SWTSirTrace {
+    /// Create a SWTSirTrace from a raw buffer and (element) length.
+    ///
+    /// The buffer is converted to a higher level representation and the buffer passed in is freed.
```

Comment:
I think the first part of this sentence is redundant, and we just need to say something like "This function will free the buffer when."

Reply:
I've just realised: there's an implicit assumption here that the buffer was allocated by `malloc` (and thus it can be freed by `free`). We should document that.